kolibri/core/assets/src/views/k-button/index.vue

```diff
@@ -0,0 +1,136 @@
+<template>
+
+  <button
+    class="k-button"
+    ref="button"
+    :class="buttonClass"
+    :type="type"
+    :disabled="disabled"
+    @click="handleClick">
+    {{ text }}
+  </button>
+
+</template>
+
+
+<script>
+
+  export default {
+    props: {
+      text: {
+        type: String,
+        required: true,
+      },
+      primary: {
+        type: Boolean,
+        default: false,
+      },
+      raised: {
+        type: Boolean,
+        default: true,
+      },
+      disabled: {
+        type: Boolean,
+        default: false,
+      },
+      type: { type: String },
+    },
+    computed: {
+      buttonClass() {
+        if (this.primary && this.raised) {
+          return 'k-button-primary-raised';
+        } else if (this.primary && !this.raised) {
+          return 'k-button-primary-flat';
+        } else if (!this.primary && this.raised) {
+          return 'k-button-secondary-raised';
+        } else if (!this.primary && !this.raised) {
+          return 'k-button-secondary-flat';
+        }
+      },
+    },
+    methods: {
+      handleClick(event) {
+        this.$emit('click', event);
+        this.$refs.button.blur();
+      },
+    },
+  };
```

Comment:
Let's add lots of comments to fill in the API docs:

![image](https://user-images.githubusercontent.com/2367265/28465967-8a668c10-6ddf-11e7-8773-b4c937b91330.png)

see `core-modal` for an example of this in action.